test/testobjects/TestOrchestrations.ts

```diff
@@ -316,10 +328,14 @@ export class TestOrchestrations {
         const retryOptions = new df.RetryOptions(100, 5);
         const output = [];
         output.push(
-            context.df.callSubOrchestratorWithRetry("SayHelloInline", retryOptions, "Tokyo")
+            context.df.callSubOrchestratorWithRetry("SayHelloInline", retryOptions, {

```

Comment:
Anyone else conflicted on `input`? Like yes, it's technically optional - but it also seems like it's an important argument to the function. How common is it to pass an input? I could see myself getting annoyed at doing `{ input: ...` all the time

Reply:
Yes, @hossam-nasr and me were just talking about this.
I do think passing inputs to sub-orchestrations is common, especially because they're used to offload work that you do not want your orchestrator to compute itself, so they usually get passed in some work batch as input.

Just to ask the obvious question: how do you feel about an overload here? After all, it is both an optional argument, but also a common one, so I think it's a fair candidate.

Reply:
Well `input` is type `unknown` which makes it really difficult to do an overload. Both `input` and `options` could be an object, so we don't really know which one a user is passing us. Technically since `input` could be anything, an options object _is_ a valid input

Reply:
Good point on the `unknown` type.
At this time - I would recommend we make an exception for the `input` parameter have it stay as a positional parameter despite being optional. I agree it's an imperfect choice, but we have [other threads](https://github.com/Azure/azure-functions-durable-js/issues/418) where improvements to this API are being discussed where I think this imperfection can be better tackled.

Reply:
Hmm I don't disagree that it's annoying to have to type `{ input: ... }` every time, especially if you're not specifying an `instanceId`, and I was conflicted about this decision too. But I saw this as just being the orchestration equivalent of `client.startNew()` which we did change. If we're going to leave this as-is, what should we do about `client.startNew()`? I think the old implementation of `client.startNew()` was especially problematic because the order of the arguments was different than here: it was the instanceId first, _then_ the input, meaning you always had to write something like: `client.startNew(name, undefined, input)` if you didn't want to specify the `instanceId`, which was recommended anyway.

If we're going to keep this as-is, should we also flip the order of arguments for `client.startNew()`?